pkg/controllers/jobset_controller.go

```diff
@@ -237,17 +234,17 @@ func (r *JobSetReconciler) calculateAndUpdateReplicatedJobsStatuses(ctx context.
 	return r.Status().Update(ctx, js)
 }
 
-func (r *JobSetReconciler) calculateReplicatedJobStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) ([]jobset.ReplicatedJobStatus, error) {
+func (r *JobSetReconciler) calculateReplicatedJobStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) []jobset.ReplicatedJobStatus {
 	log := ctrl.LoggerFrom(ctx)
 
 	// Prepare replicatedJobsReady for optimal iteration
-	replicatedJobsReady := map[string]int32{}
+	replicatedJobsReady := map[string]*[2]int32{}
```

Comment:
This is harder to follow and makes it easy to get the indices of each status wrong, just use two maps pls.

Reply:
I was thinking about memory optimization. ok.

Reply:
in the larger scheme of things, this is not significant